src/main/java/org/radarcns/management/service/MetaTokenService.java

```diff
@@ -55,16 +56,17 @@ public MetaToken save(MetaToken metaToken) {
      * @return the entity
      * @throws CustomNotFoundException if there is no project with the given id
      */
-    public TokenDTO fetchToken(String tokenName) throws CustomNotFoundException,
-            MalformedURLException {
+    public TokenDTO fetchToken(String tokenName) throws
+            MalformedURLException, CustomNotFoundException {
         log.debug("Request to get Token : {}", tokenName);
         MetaToken fetchedToken = getToken(tokenName);
 
-        if (!fetchedToken.isFetched()) {
+        // process the response if the token is not fetched or not expired
+        if (!fetchedToken.isFetched() && Instant.now().isBefore(fetchedToken.getExpiryDate())) {
```

Comment:
If expired, remove from database as well? Or is there a separate process for removing expired tokens?

Reply:
My plan was to have a scheduled job to remove expired and fetched tokens